src/chrome/internal/features/smartStep.ts

```diff
@@ -75,29 +52,31 @@ export class SmartStepLogic implements IComponent, IStackTracePresentationLogicP
         }
     }
 
-    public stepInIfOnSkippedSource(): void {
-        throw new Error('Not implemented TODO DIEGO');
+    public sendUpdatedPause(): void {
+        // TODO
+        // this._eventsToClientReporter.sendDebuggeeIsStopped({ reason: Reason})
     }
 
-    public async shouldSkip(frame: ScriptCallFrame): Promise<boolean> {
+    public async shouldSkip(paused: PausedEvent): Promise<boolean> {
         if (!this._isEnabled) return false;
 
-        const clientPath = this._pathTransformer.getClientPathFromTargetPath(frame.location.script.runtimeSource.identifier)
-            || frame.location.script.runtimeSource.identifier;
-        const mapping = await this._sourceMapTransformer.mapToAuthored(clientPath.canonicalized, frame.codeFlow.location.position.lineNumber, frame.codeFlow.location.position.columnNumber);
+        if (paused.reason !== 'other') return false;
+
+        const frame = paused.callFrames[0];
+        const mapping = await this._sourceMapTransformer.mapToAuthored(frame.location.script.url, frame.codeFlow.location.position.lineNumber, frame.codeFlow.location.position.columnNumber);
```

Comment:
Is this equivalent to frame.location.mappedToSource().resource.isMappedSource() ?

Reply:
Oh, possibly